pkg/consoleplugin/scriptserver/scriptserver.go

```diff
@@ -0,0 +1,77 @@
+package scriptserver
+
+import (
+	"embed"
+	"net/http"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sync"
+)
+
+var (
+	log = logf.Log.WithName("web_console_script_server")
+)
+
+//go:embed static/*
+var staticFiles embed.FS
+
+type ScriptServer interface {
+	HandleScriptRequest(w http.ResponseWriter, r *http.Request)
+}
+
+type scriptServer struct {
+	rw    sync.RWMutex
+	cache map[string][]byte
+}
+
+func NewScriptServer() ScriptServer {
+	return &scriptServer{
+		cache: map[string][]byte{},
+	}
+}
+
+func (s *scriptServer) HandleScriptRequest(w http.ResponseWriter, r *http.Request) {
+	data, err := s.loadResource(r.RequestURI)
+	if err != nil {
+		w.WriteHeader(http.StatusNotFound)
+		log.Error(err, "error while loading resource")
+	}
+
+	w.WriteHeader(http.StatusOK)
+	if _, err := w.Write(data); err != nil {
+		log.Error(err, "unable to write response")
+	}
+}
+
+func (s *scriptServer) loadResource(path string) ([]byte, error) {
+	data := s.loadResourceFromCache(path)
+	if data != nil {
+		return data, nil
+	}
+
+	return s.validateCachedResource(path)
+}
+
+func (s *scriptServer) loadResourceFromCache(path string) []byte {
+	s.rw.RLock()
+	defer s.rw.RUnlock()
+
+	if val, ok := s.cache[path]; ok {
+		return val
+	}
+
+	return nil
+}
+
+func (s *scriptServer) validateCachedResource(path string) ([]byte, error) {
+	s.rw.Lock()
+	defer s.rw.Unlock()
+
+	fileData, err := staticFiles.ReadFile("static" + path)
+	if err != nil {
+		return nil, err
+	}
+
+	s.cache[path] = fileData
+
+	return fileData, nil
+}
```

Comment:
We will almost certainly need to perform some string replacement for the file contents so this is in preparation for that.

Reply:
Just for the context. We need to insert the Pendo key into the JS file. The key is supposed to be loaded from our configuration. Currently it's hardcoded: https://github.com/codeready-toolchain/member-operator/pull/417/commits/6f109baabe662a310f3d944836c0dfcff8ddf0d2 But we need to move it start using configuration in this PR.

Ideally we should apply the current configuration without restarting the service but it's fine for now to load configuration when the service is being started IMO. We can improve later.

Reply:
Just to add to this, it's non-trivial to populate a map at startup time because the actual content filenames are dynamic, for example `exposed-pendo-chunk-0f1e4648dfea45bc1052.min.js`.  This filename would change if we were to regenerate it after making changes to the source JS, meaning we would have to then update the populating of the map cache code, thereby introducing a vector for human error occurring.

Reply:
> Ideally we should apply the current configuration without restarting the service but it's fine for now to load configuration when the service is being started IMO. We can improve later.

@alexeykazakov in this case, you would need to watch the MemberConfig CR in the service and invalidate the cache when the key is changed - this seems to be a kind of overkill for such a small use case. Instead of it, I would rather set the key as a variable in the Deployment manifest of the service and let the member-operator redeploy the service when the key is changed.

>  This filename would change if we were to regenerate it after making changes to the source JS

@sbryzak if the filename is changed, then it needs to be one in our GH repository, which means:
1. making a PR
2. mering it 
3. releasing a new version of member operator including this service
4. redeploying the service with a new tag that contains the modified/renamed filename

> meaning we would have to then update the populating of the map cache code, thereby introducing a vector for human error occurring.

not at all - this will be fully automatic as described above :wink: